test/language/identifiers/part-zwj-escaped.js

```diff
@@ -0,0 +1,10 @@
+// Copyright (C) 2020 Adam Kluball. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Adam Kluball
+esid: sec-names-and-keywords
+description: zero width joiner is a valid identifier part
+---*/
+
+var $\u200D;
```

Comment:
I believe for this specific case we need to improve this test a little bit to avoid false positives, such as an implementation taking `\u200D` and `\u200C` as white space, supposedly. 

Maybe this can be done w/ a new file leaving the current one as is, or just blending this file with the `test/language/identifiers/part-zwnj-escaped.js`. I prefer the former option, creating a new test file to distinguish those identifiers.

```suggestion
var $ = 1;
var $\u200D = 2;
var $\u200C = 3;

assert.sameValue($, 1);
assert.sameValue($\u200D, 2);
assert.sameValue($\u200C, 3);
```

Reply:
Added an assingment test. It seems like it makes the other tests redundant. I could remove them and rename the assignment test to part-zwj-zwnj-unescaped.js.

Reply:
Thanks! I believe it's fine either way. Just let me know when it's ready.

Reply:
It's ready